plugins/nf-azure/src/main/nextflow/cloud/azure/fusion/AzFusionEnv.groovy

```diff
@@ -48,13 +48,13 @@ class AzFusionEnv implements FusionEnv {
         // the account name is always required
         result.AZURE_STORAGE_ACCOUNT = cfg.storage().accountName
 
-        // If a Managed Identity or Service Principal is configured, Fusion only needs to know the account name
-        if (cfg.managedIdentity().isConfigured() || cfg.activeDirectory().isConfigured()) {
+        // If a Managed Identity is configured, Fusion only needs to know the account name
+        if (cfg.managedIdentity().isConfigured()) {
             return result
         }
 
         // If a SAS token is configured, instead, Fusion also requires the token value
```

Comment:
@adamrtalbot Could we please update this comment to something like:
```
        // Otherwise, Fusion also requires a SAS Token
        // (yes, even if `cfg.storage().sasToken` has not been explicitly defined)
```
so that this doesn't happen again?

Reply:
That's not much different from what the code itself already tells. I would prefer to have a explicit comment telling us why it's needed despite the configuration not requesting it explicitly.